pkg/volume/csi/csi_plugin.go

```diff
@@ -205,16 +205,15 @@ func (p *csiPlugin) Init(host volume.VolumeHost) error {
 	p.host = host
 
 	if utilfeature.DefaultFeatureGate.Enabled(features.CSIDriverRegistry) {
-		csiClient := host.GetCSIClient()
+		csiClient := host.GetKubeClient()
 		if csiClient == nil {
-			klog.Warning("The client for CSI Custom Resources is not available, skipping informer initialization")
-		} else {
-			// Start informer for CSIDrivers.
-			factory := csiapiinformer.NewSharedInformerFactory(csiClient, csiResyncPeriod)
-			p.csiDriverInformer = factory.Csi().V1alpha1().CSIDrivers()
-			p.csiDriverLister = p.csiDriverInformer.Lister()
-			go factory.Start(wait.NeverStop)
+			return errors.New("unable to get Kubernetes client")
 		}
+		// Start informer for CSIDrivers.
+		factory := csiapiinformer.NewSharedInformerFactory(csiClient, csiResyncPeriod)
```

Comment:
Now that we're using core APIs, we should be able to use the shared informer instead of creating a new shared informer factory.  I'm ok with refactoring this in a separate pr so that we can get the API published and sidecar dependencies updated.

Reply:
Will do in a separate PR.